lib/improver/utilities/pad_spatial.py

```diff
@@ -115,12 +115,12 @@ def create_cube_with_halo(cube, halo_radius):
         halo_cube (iris.cube.Cube):
             New cube defining the halo-padded grid (data set to zero)
     """
-    halo_size_x, halo_size_y = convert_distance_into_number_of_grid_cells(
+    halo_size = convert_distance_into_number_of_grid_cells(
```

Comment:
Why are we losing the ability to define a halo for a non-square grid cell grid? I would suggest we calculate ```halo_size_x``` and ```halo_size_y``` and maintain this functionality as it costs essentially nothing.